### PR TITLE
179 saved search updates

### LINKED ISSUE #179: [Bug] pg_size and pg_num have no effect in the saved search execute method
### Description

The `execute()` method for saved searches is not doing anything with `pg_num` or `pg_size`.

### Steps to Reproduce

1. Try using `pg_num` or `pg_size` on a saved search `execute()` call.
2. In debug mode, notice that the default params are always in the request
3. Also, look at the code and see that they are not used.

### Expected Behavior
To be able to use the `pg_size` and `pg_num` parameters when executing a saved search

### Actual Behavior
The default values for the query are always used.

### PR DESCRIPTION
**Pull request by DiscoRiver**
### Description of Change

This code fixes #179;

- `py42.sdk.queries.BaseQuery.from_dict` now accepts keyword args, which are included in the returning class object. In this case they are used in `FileEventQuery`
- `SavedSearchClient.get_query` now accepts `pg_num` and `pg_size` keyword arguments.
- `SavedSearchClient.execute` now works with user-defined `pg_num` and `pg_size` arguments
- Added new `py42.settings` variable `security_events_per_page`
- Added priority for parameters passed directly to `execute()`, to override `security_events_per_page` for specific calls.
- Added two tests for this change, `test_execute_calls_post_with_expected_setting_page_param`, and `test_execute_calls_post_with_expected_page_params`, which test the new `setting.security_events_per_page`, and `pg_num`/`pg_size` parameters.

### Issues Resolved

- fixes #179

### Testing Procedure

To test pages are returned, and error `400` is handled appropriately (since `SavedSearchService.execute` doesn't return a generator), the following code can be used to test. The `pg_size` parameter can also be passed to `execute()`.

```
import py42.sdk 
import py42.settings as settings
from py42.exceptions import Py42BadRequestError
import urllib3

urllib3.disable_warnings()

settings.verify_ssl_certs = False
settings.security_events_per_page = 100

sdk = py42.sdk.from_local_account("https://console.us.code42.com", "username", "password")

pg_num = 1
while True:
    try:
        resp = sdk.securitydata.savedsearches.execute("saved-search-id", pg_num=pg_num)
    except Py42BadRequestError:
        print("Got all {} pages!".format(pg_num))
        break
    pg_num+=1
```

### CODE REVIEW

File: src/py42/services/savedsearch.py [hunk 1]
```diff
@@ -32,19 +32,21 @@ def get_by_id(self, search_id):
         uri = u"{}/{}".format(self._resource, search_id)
         return self._connection.get(uri)
 
-    def get_query(self, search_id):
+    def get_query(self, search_id, pg_num=None, pg_size=None):
```
Comment (alanag13) [inline src/py42/services/savedsearch.py:35]: for consistency with other methods, this should be `page_size` and `page_num`. The base class is also looking for keyword arguments by these names.

File: src/py42/services/savedsearch.py [hunk 2]
```diff
@@ -32,19 +32,21 @@ def get_by_id(self, search_id):
...
         Returns:
             :class:`py42.sdk.queries.fileevents.file_event_query.FileEventQuery`
         """
         response = self.get_by_id(search_id)
         search = response[u"searches"][0]
-        return FileEventQuery.from_dict(search)
+        return FileEventQuery.from_dict(search, page_number=pg_num, page_size=pg_size)
 
-    def execute(self, search_id, pg_num=1, pg_size=10000):
+    def execute(self, search_id, pg_num=None, pg_size=None):
```
Comment (alanag13) [inline src/py42/services/savedsearch.py:49]: `page_size`, `page_num`

File: src/py42/services/savedsearch.py [hunk 3]
```diff
@@ -55,5 +57,5 @@ def execute(self, search_id, pg_num=1, pg_size=10000):
         Returns:
             :class:`py42.response.Py42Response`
         """
-        query = self.get_query(search_id)
+        query = self.get_query(search_id, pg_num=pg_num, pg_size=pg_size)
```
Comment (alanag13) [inline src/py42/services/savedsearch.py:60]: `page_size`, `page_num`

File: src/py42/services/savedsearch.py (hunk 1 above)
Comment (DiscoRiver) [reply src/py42/services/savedsearch.py:35]: To clarify, is this `page_num` or `page_number`? `BaseQuery` is `page_number` that's why I ask.

Comment (unparalleled-js) [reply src/py42/services/savedsearch.py:35]: @DiscoRiver  You correct that it is `page_number` in the query classes... I have `page_num` in other parts of the code base though.  Perhaps we should make a plan to standardize these.

File: src/py42/services/savedsearch.py (hunk 2 above)
Comment (unparalleled-js) [reply src/py42/services/savedsearch.py:49]: As pointed out above, it is `page_number` in the query classes

File: src/py42/services/savedsearch.py (hunk 1 above)
Comment (DiscoRiver) [reply src/py42/services/savedsearch.py:35]: Sounds good. I'll change it to `page_number` for now, but I can raise an issue for standardizing it (happy to do the work on that too).

Comment (unparalleled-js) [reply src/py42/services/savedsearch.py]: I am delighted at how much you are wanting to work on py42! Much appreciated. I think we should save the page num standardization for a greater release, probably like a 2.0, but if it is something you want to take on, I will try to make that happen when that time comes.

File: CHANGELOG.md [hunk 4]
```diff
@@ -35,6 +35,13 @@ how a consumer would use the library (e.g. adding unit tests, updating documenta
 
 ### Added
 
+- Added additional user-adjustable setting for security events page size:
+    - `py42.settings.security_events_per_page`
+
+- Page pg_num and pg_size parameters for saved search queries:
```
Comment (unparalleled-js) [inline CHANGELOG.md:41]: Don't forget to change these as well

File: src/py42/services/savedsearch.py [hunk 5]
```diff
@@ -32,28 +32,32 @@ def get_by_id(self, search_id):
         uri = u"{}/{}".format(self._resource, search_id)
         return self._connection.get(uri)
 
-    def get_query(self, search_id):
+    def get_query(self, search_id, page_number=None, page_size=None):
         """Get the saved search in form of a query(`py42.sdk.queries.fileevents.file_event_query`).
 
         Args:
             search_id (str): Unique search Id of the saved search.
+            page_number (int, optional): The consecutive group of results of size pg_size in the result set to return. Defaults to 1.
```
Comment (unparalleled-js) [inline src/py42/services/savedsearch.py:40]: Still `pg_size` in the description of `page_number`

File: src/py42/services/savedsearch.py [hunk 6]
```diff
@@ -32,28 +32,32 @@ def get_by_id(self, search_id):
...
+            page_size (int, optional): The maximum number of results to be returned. Defaults to 10,000.
```
Comment (unparalleled-js) [inline src/py42/services/savedsearch.py:41]: Also, these both should say `Defaults to None` now

File: src/py42/services/savedsearch.py [hunk 7]
```diff
@@ -32,28 +32,32 @@ def get_by_id(self, search_id):
...
-    def execute(self, search_id, pg_num=1, pg_size=10000):
+    def execute(self, search_id, page_number=None, page_size=None):
         """
         Execute a saved search for given search Id and return its results.
 
         Args:
             search_id (str): Unique search Id of the saved search.
-            pg_num (int, optional): The consecutive group of results of size pg_size in the result set to return. Defaults to 1.
-            pg_size (int, optional): The maximum number of results to be returned. Defaults to 10,000.
+            page_number (int, optional): The consecutive group of results of size pg_size in the result set to return. Defaults to 1.
```
Comment (unparalleled-js) [inline src/py42/services/savedsearch.py:57]: `pg_size` in the description for `page_number`

Comment (unparalleled-js) [reply src/py42/services/savedsearch.py:57]: Also, the defaults need to be updated

**Review by unparalleled-js — APPROVED** [review]

**Pull request merged**